### PR TITLE
Add SNV as sub-command to the command line and handle errors related to empty windows

### PR DESCRIPTION
**Pull request by sposadac**

### CODE REVIEW

File: src/cpp/dpm_sampler.cpp [hunk 1]
```diff
@@ -526,7 +526,7 @@ void read_data(char* filename, std::ofstream& out_file)
     unsigned int seq = 0, i, j = 0, k = 0;
     out_file << "# reading data\n";
     if ((data = fopen(filename, "r")) == NULL) {
-        out_file << "Impossible to open file " << filename << "\n";
+        out_file << "Impossible to open file " << filename << std::endl;
```
Comment (SoapZA) [inline src/cpp/dpm_sampler.cpp]: any reason you need `std::endl` here?

Comment (sposadac) [reply src/cpp/dpm_sampler.cpp]: I don't think so, and since I didn't change them all I can't argue about the style.

**Comment by ozagordi** [comment]
Looks good to me as well. This would help prevent quite a number of issues from our user base.
Thanks @sposadac!!!

File: src/cpp/dpm_sampler.cpp (hunk 1 above)
Comment (SoapZA) [reply src/cpp/dpm_sampler.cpp]: Can you please change to `'\n'`?

Comment (sposadac) [reply src/cpp/dpm_sampler.cpp]: Done!

File: src/cpp/dpm_sampler.cpp [hunk 2]
```diff
@@ -558,7 +558,7 @@ void read_data(char* filename, std::ofstream& out_file)
     fclose(data);
 
     if (n == 1) {
-        out_file << "Nothing to do, have only one read... (n == 1)\n";
+        out_file << "Nothing to do, have only one read... (n == 1)" << std::endl;
```
Comment (SoapZA) [inline src/cpp/dpm_sampler.cpp]: still and `endl` here

Comment (sposadac) [reply src/cpp/dpm_sampler.cpp]: Yes, this one I put on purpose to flush to output.

Comment (sposadac) [reply src/cpp/dpm_sampler.cpp]: Yes, this one I put on purpose to flush to output.

Comment (SoapZA) [reply src/cpp/dpm_sampler.cpp]: alrighty then

**Comment by SoapZA** [comment]
go ahead and mere then

**Pull request merged**